simulation/traffic_simulator/src/hdmap_utils/hdmap_utils.cpp

```diff
@@ -1455,18 +1491,63 @@ auto HdMapUtils::getLongitudinalDistance(
       return to.s - from.s;
     }
   }
-  const auto route = getRoute(from.lanelet_id, to.lanelet_id);
+  const auto route = getRoute(from.lanelet_id, to.lanelet_id, allow_lane_change);
   if (route.empty()) {
     return std::nullopt;
   }
   double distance = 0;
-  for (const auto lanelet_id : route) {
-    if (lanelet_id == from.lanelet_id) {
-      distance = getLaneletLength(from.lanelet_id) - from.s;
-    } else if (lanelet_id == to.lanelet_id) {
-      distance = distance + to.s;
+
+  auto with_lane_change = [this](
+                            const bool allow_lane_change, const lanelet::Id current_lanelet,
+                            const lanelet::Id next_lanelet) -> bool {
+    if (allow_lane_change) {
+      auto next_lanelet_ids = getNextLaneletIds(current_lanelet);
+      auto next_lanelet_itr = std::find_if(
+        next_lanelet_ids.begin(), next_lanelet_ids.end(),
+        [next_lanelet](const lanelet::Id & id) { return id == next_lanelet; });
+      return next_lanelet_itr == next_lanelet_ids.end();
+    } else {
+      return false;
+    }
+  };
+  for (unsigned int i = 0; i < route.size(); i++) {
+    if (i < route.size() - 1 && with_lane_change(allow_lane_change, route[i], route[i + 1])) {
```

Comment:
I suggest to use

```
  for (auto itr = std::adjacent_find(route.begin(), route.end(),
         [](const auto &, const auto &) { return true; });
```

Reply:
I don't think `std::adjacent_find` is match for this usecase and introduce bug especially with multiple lane change route.

Reply:
Sorry, I misunderstood the usage of std::adjacent_find

```
if (auto itr = std::adjacent_find(route.begin(), route.end(),
       [](const auto & waypoint_a, const auto & waypoint_b) { `Functions return true if the ` }; 
   itr != route.end()){
   }
```
```